### ISSUE TITLE
TBDs not working outside of root of project

### ISSUE THREAD

**Issue by mrRachar** [mrRachar/AElfi#2]
Template Based Documents do not seem to work when outside the project's root. Its seems as if they are redirected through `env.py`. I don't believe there is a problem with other Python files. The output at first is that access is forbidden, but after the removal of `_aelfi` folder protection, this is changed to an error from `env.py`, saying 'config.json' doesn't exist. This is probably due to it trying to run `tbd.py` with the `_aelfi` folder as cwd. All this seems to prove the hypothesis. It seems like changes are needed to the build file's `.htaccess` output, and as this problem _does not appear with the older .conf system_ it may provide useful insights into solving this.

**Comment by mrRachar** [comment]
In a test `.htaccess` file, with only the tbd rule, it works fine:

``` htaccess
Require all granted
Options +ExecCGI -Indexes
DirectoryIndex index.py index.pyv index.php index.html index.htm index.xml inde$
RewriteEngine on

# _aelfi/tbd.py?AELFI_TBD_PAGE={1}
RewriteCond %{REQUEST_FILENAME}  -f
RewriteCond %{REQUEST_FILENAME} "\.pyv$"
RewriteRule "^(.*)$" "_aelfi/tbd\.py?AELFI_TBD_PAGE=$1" [QSA,L,END]
```

**Comment by mrRachar** [comment]
It works on non-index pyv files, and the line `RewriteCond %{REQUEST_FILENAME} !"^/var/www/html/test/rnb/_aelfi/?"` doesn't seem to change anything.

**Comment by mrRachar** [comment]
Partial fix: correct all to `pageloc` or `page_location` (incl. Request.directory property). Doesn't stop .htaccess file, but fixes whole working directory problem

**Comment by mrRachar** [comment]
4e9409c0a6de33007d35f8102963c1b0df8c4f91 provides a fix to both the `pageloc/page_location` and the .htaccess problems, which should work. This hasn't been fully tested, we need to see what scope the `.pyv` files end up in, including imports/templates etc. which may not be relative to the right place. It also doesn't fix it in the best way, and doesn't work when the aelfi folder is protected. Was is this stuff???

**Comment by mrRachar** [comment]
08ea487aca4bf3d4868974ea9745e2c187f0cd05 provides a more proper, tradition fix, but `_aelfi' folder protection still seems problematic, and where the tbd file's cwd is has yet to be tested. Once it has however, the new htaccess-fix branch will be merged in

**Comment by mrRachar** [comment]
Adding the following lines just before the aelfi loader rule protects most of the `_aelfi` folder, but still leaves the programme open to requesting superfolders, which is _very open to bugs_

``` htaccess
RewriteCond %{REQUEST_FILENAME} "^/var/www/html/test/snb/_aelfi/[^l]"
RewriteRule ".*" - [F]
```

See #3 for more info on this security flaw

**Comment by mrRachar** [comment]
This issue seems resolved by 08ea487, and 3241e500ac9425e33c305f984e781d4d3ec91c21 has fixed the security flaw, see #3 for more.
